### PR TITLE
Added idletiming

### PR DESCRIPTION
**Pull request by oxtoacart**
For getlantern/lantern-internal#102

Since idletiming works well on our proxies, I went ahead and just used it here too.

### CODE REVIEW

File: tlsproxy.go [hunk 1]
```diff
@@ -128,18 +139,20 @@ func doRun(listen func() (net.Listener, error), dial func() (net.Conn, error)) {
 	log.Debugf("Listening for incoming connections at: %v", l.Addr())
 
 	for {
-		in, err := l.Accept()
+		_in, err := l.Accept()
 		if err != nil {
 			log.Fatalf("Unable to accept: %v", err)
 		}
 
+		in := idletiming.Conn(_in, *idleTimeout, nil)
```
Comment (joesis) [inline tlsproxy.go]: idletiming creates a goroutine and a timer, which may have some performance penalty I guess? Do you think it's useful to benchmark it against a version that using `SetKeepAlive` and `SetKeepAlivePeriod`?

Comment (oxtoacart) [inline tlsproxy.go]: Let me try benchmarking against just a version without any idletimeout.

Comment (joesis) [inline tlsproxy.go]: Thinking again, it should be neglectable compared to the CPU required to handle TLS and copy data.

File: tlsproxy_benchmark_test.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	go runServer(sl, l.Addr().String(), idleTimeout, serverConfig)
+	go runClient(cl, sl.Addr().String(), idleTimeout, clientConfig)
+
+	clientAddr := cl.Addr().String()
+
+	b.ResetTimer()
+
+	buf := make([]byte, len(data))
+	for i := 0; i < b.N; i++ {
+		conn, err := net.Dial("tcp", clientAddr)
```
Comment (joesis) [inline tlsproxy_benchmark_test.go]: I tried to run the loop inside goroutine to see how it work concurrently, but tlsproxy failed to accept with "use of closed network connection".

It still won't reflect the real performance shape running in production, though, so let's ignore it.

**Comment by joesis** [comment]
LGTM. The benchmarks are comparable w/ or w/o idletiming. Merging.

**Comment by oxtoacart** [comment]
Thanks @joesis !

**Pull request merged**